app/models/quote.rb

```diff
@@ -3,22 +3,37 @@ class Quote < ActiveRecord::Base
   acts_as_paranoid
   belongs_to :user
   belongs_to :reporter, class_name: "User"
-  validates :user, presence: true
-  validates :reporter, presence: true
   validates :text, presence: true
   serialize :text
   validate :reporter_and_user_differ
+  after_save :destroy_history
 
   encrypts :text
 
   scope :ordered, -> { order('created_at DESC') }
   scope :random, -> { order('RAND()') }
   scope :with_user, -> { includes(:user) }
 
+
+  def anonymous?
+    user.nil?
+  end
+
+  def destroy_history
+    return unless anonymous?
+    versions.destroy_all
+  end
+
+  def name
+    return 'Anoniem' if anonymous?
+
+    user.name
+  end
+
   private
 
   def reporter_and_user_differ
-    return unless reporter_id == user_id
+    return if user_id.nil? || user_id != reporter_id
 
     errors.add(:user, ": Je kan toch niet jezelf quoten!")
```

Comment:
IMO iets logischer te volgen, hoeft niet per se 

```suggestion
    return if user_id.nil?

    errors.add(:user, ": Je kan toch niet jezelf quoten!") if reporter_id == user_id
```

Reply:
Persoonlijk minder fan van omdat je juist je statement opsplitst terwijl beide nu invloed hebben daarop. Ik maak wel van `user_id.nil?` -> `anonymous?`, denk dat dat het nog een stukje logischer maakt

Reply:
Ik vind het juist logischer opgebroken omdat eerst check je of je de validatie moet dan en dan doe je de validatie ipv daar 1 statement van te maken. 

`anonymous?` -> `nil?` imo dan leg je meteen uit waarom hij nil is in sommige gevallen 👍